test/expression/parser_test.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:angular_ast/src/expression/parser.dart';
+import 'package:test/test.dart';
+
+void main() {
+  test('should parse non-pipe expressions', () {
+    const [
+      'foo',
+      'foo + bar',
+      'foo(bar)',
+      'foo ? bar : baz',
+      'foo?.bar?.baz',
+      'foo & bar',
+      'foo ?? bar',
+      'foo ??= bar',
+      'foo == bar',
+      'foo || bar',
+    ].forEach((expression) {
+      expect(parseExpression(expression).toSource(), expression);
+    });
+  });
+
+  test('should parse a simple pipe', () {
+    expect(
+      parseExpression('foo | bar').toSource(),
+      r'$$ng.pipe.bar(foo)',
```

Comment:
Ah, I see what you're doing here.

I think in this case you should make sure the parser only accepts identifiers as a pipename, rather than...I think the current implementation accepts BinaryXorExpression (which could mean `$ng.pipe.(any + stuff + in + here)(foo)`)